test/parallel/test-async-wrap-constructor.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const async_hooks = require('async_hooks');
+
+assert.throws(() => {
```

Comment:
@Shadowbeetle you gave `false` as a previously failing argument. Could you add a set of test with `false`?
Also this can be done in a double loop:
```js
for (let badArg of [1, false, true, null, 'hello']) {
  for (let field of ['init', 'before', 'after', 'destroy']) {
    assert.throws(() => {
      async_hooks.createHook({ [field]: badArg });
    }, /^TypeError: before must be a function$/);
  }
}